python/tank/context.py

```diff
@@ -1043,15 +1094,16 @@ def from_entity(tk, entity_type, entity_id):
         "step": None,
         "user": None,
         "task": None,
-        "additional_entities": []
+        "additional_entities": [],
+        "source_entity": source_entity,
     }
 
     if entity_type == "Task":
         # For tasks get data from shotgun query
         task_context = _task_from_sg(tk, entity_id)
         context.update(task_context)
 
-    elif entity_type in ["PublishedFile", "TankPublishedFile"]:
+    elif entity_type in ["PublishedFile", "TankPublishedFile", "Version"]:
```

Comment:
this is great! I think we may have a separate ticket for this - please do a quick check and perhaps we can close that out at the same time. I think we should also add `Cut` since it shares the same conceptual semantics.